Decompiler/Program.cs

```diff
@@ -621,41 +641,85 @@ private static void DumpVPK(Package package, string type, string newType)
 
                 filePath = FixPathSlahes(filePath);
 
+                if (!string.IsNullOrEmpty(Options.Filter) && !filePath.StartsWith(Options.Filter, StringComparison.Ordinal))
+                {
+                    continue;
+                }
+                
                 if (Options.OutputFile != null)
                 {
-                    if (OldPakManifest.TryGetValue(filePath, out uint oldCrc32) && oldCrc32 == file.CRC32)
+                    if (OldPakManifest.TryGetValue(filePath, out oldCrc32) && oldCrc32 == file.CRC32)
                     {
                         continue;
                     }
 
                     OldPakManifest[filePath] = file.CRC32;
                 }
 
-                Console.WriteLine("\t[archive index: {0:D3}] {1}", file.ArchiveIndex, filePath);
+                if (!Options.Silent) Console.WriteLine("\t[archive index: {0:D3}] {1}", file.ArchiveIndex, filePath);
 
                 byte[] output;
                 package.ReadEntry(file, out output);
 
-                if (type.EndsWith("_c", StringComparison.Ordinal))
+                if (type.EndsWith("_c", StringComparison.Ordinal) && !Options.Extract)
                 {
                     using (var resource = new Resource())
                     {
                         using (var memory = new MemoryStream(output))
                         {
                             resource.Read(memory);
-                        }
```

Comment:
It is necessary for extracting sound resources, as the memory memorystream would be closed and GetSound() would fail. I guess you would prefer creating a new stream but that means more code. 
There is no harm in keeping the stream open until work with the resource is done?!